subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/VersionConflictResolutionIntegrationTest.groovy

```diff
@@ -636,6 +636,12 @@ repositories {
 dependencies {
     compile "org:other:1.7"
 }
+
+configurations.all {
+    resolutionStrategy.dependencySubstitution {
+        substitute project('test') with module('org:test:2.1')
+    }
+}
```

Comment:
This effectively breaks the use case described in GRADLE-2920, where the user wants to find the latest published version of the current module. To make this work, the test needs to demonstrate that this version can be _discovered_, not just resolved.